### PR TITLE
refactor: support 1.0.0

### PR DESCRIPTION
**Pull request by caspiano**
Converts from `JSON.mapping` to `JSON::Serializable` (thanks vim macros)

### CODE REVIEW

**Comment by caspiano** [comment]
Closes #6

**Comment by stakach** [comment]
should we move this to spider-gazelle or place-labs?

**Comment by kimburgess** [comment]
Moving to the place-labs org sounds like a good idea. It's been unmaintained for a while as was originally put together for https://github.com/aca-labs/bob, but I'm assuming this revival means it still has life/use in other components?

File: src/docker/api/models/response.cr [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+module Docker::Api::Models
+  # :nodoc:
+  abstract struct Response
+    include JSON::Serializable
+
+    macro extended
+      macro finished
+        # Initializer based off {{@type.id}} properties
+        def initialize(
```
Comment (kimburgess) [inline src/docker/api/models/response.cr:11]: This _may_ cause some odd behaviour if types extending this are then extended themselves. See https://forum.crystal-lang.org/t/constructor-variants-and-inheritance/3310/3.

**Review by kimburgess — APPROVED** [review]
Definitely some much needed attention! Looks good aside from comment below.

There's a bunch of repetition in those key transforms, but think we should be able to solve that more generally rather than doing something custom just in this lib. Just opened https://github.com/crystal-lang/crystal/issues/10793 to discuss that.

**Pull request merged**